crates/biome_js_analyze/src/lint/nursery/use_import_extensions.rs

```diff
@@ -88,7 +88,11 @@ impl Rule for UseImportExtensions {
     fn run(ctx: &RuleContext<Self>) -> Self::Signals {
         let node = ctx.query();
 
-        let file_ext = ctx.file_path().extension()?.to_str()?;
+        let file_ext = ctx
+            .file_path()
+            .extension()
+            .and_then(|ext| ext.to_str())
+            .unwrap_or("js");
```

Comment:
@ematipico docs generation failed because it seems there is no file extension during it. I added fallback which seems reasonable in this case, but that should prob be investigated further.

Reply:
Oh, right. This must be fixed inside the `rules_check/lib.rs` file, but I believe @Sec-ant has done some improvements there. 

Do you mind adding a comment saying that you're adding a fallback because of flaky tooling, and that should not happen in production?

Reply:
It's fixed. The `rules_check` now uses the code block tag (`js` here) as the extension. You can remove the workaround and rebase.

Reply:
But wait, maybe we want to keep the fallback because the rule can be triggered in an unsaved file which doesn't have an extension?

Reply:
I think this will never happen in production, because clients must go through `WorkspaceServer`, which makes sure that only files with a correct `DocumentFileSource` can be handled. However, it was happening in `rules_check` because we bypassed a lot of checks, and we were not passing all the data (that you fixed)

Reply:
Ok, just tested it. Unsaved files and those with unknown extensions will not trigger this rule.

Reply:
I merged latest main and removed fallback extension. `cargo run -p rules_check` fails again, so there are still some issues.

Reply:
> I merged latest main and removed fallback extension. `cargo run -p rules_check` fails again, so there are still some issues.

Sorry, I fixed it directly on your branch.